lms/djangoapps/discussion/rest_api/discussions_notifications.py

```diff
@@ -447,7 +448,10 @@ def clean_thread_html_body(html_body):
     """
     Get post body with tags removed and limited to 500 characters
     """
-    html_body = BeautifulSoup(Truncator(html_body).chars(500, html=True), 'html.parser')
+    truncated_body = Truncator(html_body).chars(500, html=True)
+    # TODO: Remove unescape once upgraded to Django 5 and fix test_html_tags_removal by replacing " with &quot;
+    truncated_body = html.unescape(truncated_body)
```

Comment:
Is there a way to not use unescape?
Looking into why `html.escape` is becoming problem. The way I see it that the testcase failed with a difference in `script` tag.

Reply:
Why are we putting it off for the future?

Reply:
There is a way not to use `unescape`, for that we need to update the test `expected_output` based on the Django version.
Due to a security fix(mentioned in description), replaced `"` with `&quot;` in Django 5.0, we should keep our implementation according to the upgraded version and update `expected_output` once upgraded to 5.2 as mentioned in TODO.

Reply:
Yes, can we do this way, add if based on Django version in the test.

Plus can you also check what will be the effect of keeping `&quot;` instead of `"`.